search/searcher/search_similarity.go

```diff
@@ -0,0 +1,82 @@
+package searcher
+
+import (
+	"context"
+
+	"github.com/blevesearch/bleve/v2/mapping"
+	"github.com/blevesearch/bleve/v2/search"
+	"github.com/blevesearch/bleve/v2/search/scorer"
+	index "github.com/blevesearch/bleve_index_api"
+)
+
+type SimilaritySearcher struct {
+	field       string
+	vector      []float32
+	indexReader index.IndexReader
+	scorer      *scorer.ConstantScorer
+	count       uint64
+}
+
+func NewSimilaritySearcher(ctx context.Context, i index.IndexReader, m mapping.IndexMapping,
+	options search.SearcherOptions, field string, vector []float32) (search.Searcher, error) {
+	return &SimilaritySearcher{
+		indexReader: i,
+		field:       field,
+		vector:      vector,
+	}, nil
+}
+
+// need to implement methods Advance, Next, etc.
+// no-ops initially.
+
+func (s *SimilaritySearcher) Advance(ctx *search.SearchContext, ID index.IndexInternalID) (
+	*search.DocumentMatch, error) {
+	return nil, nil
+}
+
+func (s *SimilaritySearcher) Close() error {
+	return nil
+}
+
+// JUST DUMMY, NO-OP
+func (s *SimilaritySearcher) Count() uint64 {
+	return 0
+}
+
+// TODO JUST A NO-OP
+func (s *SimilaritySearcher) DocumentMatchPoolSize() int {
+	return 1
+}
+
+// TODO What does this mean?
+func (s *SimilaritySearcher) Min() int {
+	return 0
+}
+
+func (s *SimilaritySearcher) Next(ctx *search.SearchContext) (*search.DocumentMatch, error) {
+	// how to get doc matches here?
+
```

Comment:
look at code around optimizing conjunctions -> should not push similarity query lower in the hierarchy.